src/resources/styles/_animations.scss

```
@@ -0,0 +1,26 @@
+// Gradient
+@mixin gradient-transition($newGradient, $duration: .3s) {
+  background-size: 100%;
+  z-index: 1;
+
+  &:before {
+    height: 100%;
+    width: 100%;
+    position: absolute;
+    top: 0;
+    left: 0;
+    display: block;
+    background: $newGradient;
+    border-radius: inherit;
+    content: '';
+    opacity: 0;
+    z-index: -1;
+    transition: opacity $duration;
+  }
+
+  &:hover {
+    &:before {
+      opacity: 1;
+    }
+  }
```

IMHO this way is clearer:

```suggestion
  &:hover::before {
    opacity: 1;
  }
```

---

Agreed.